echopype/convert/set_groups_base.py

```diff
@@ -362,3 +362,143 @@ def beam_groups_to_convention(
 
         self._add_ping_time_dim(ds, beam_ping_time_names, ping_time_only_names)
         self._add_beam_dim(ds, beam_only_names, beam_ping_time_names)
+
+    def _add_index_data_to_platform_ds(
+        self,
+        platform_ds: xr.Dataset,
+    ) -> xr.Dataset:
+        """
+        Append index data from `.IDX` file to the `Platform` dataset.
+        Index file data contains latitude, longitude, and vessel distance traveled.
+
+        Parameters
+        ----------
+        platform_ds : xr.Dataset
+            `Platform` dataset without `.IDX` data.
+
+        Returns
+        -------
+        platform_ds : xr.Dataset
+            `Platform` dataset with `.IDX` data.
+            Contains new `time3` dimension to correspond with `.IDX` timestamps that
+            align with `vessel_distance`, `latitude_idx`, and `longitude_idx`.
+
+        Notes
+        -----
+        This function is only called for EK60/EK80 conversion.
+        """
+        timestamp_array, _, _ = xr.coding.times.encode_cf_datetime(
+            np.array(self.parser_obj.idx["timestamp"]),
+            **{
+                "units": DEFAULT_TIME_ENCODING["units"],
+                "calendar": DEFAULT_TIME_ENCODING["calendar"],
+            },
+        )
+        platform_ds = platform_ds.assign(
+            {
+                "vessel_distance": xr.DataArray(
+                    np.array(self.parser_obj.idx["vessel_distance"]),
+                    dims=("time3"),
+                    coords={"time3": timestamp_array},
+                ),
+                "latitude_idx": xr.DataArray(
+                    np.array(self.parser_obj.idx["latitude_idx"]),
+                    dims=("time3"),
+                    coords={"time3": timestamp_array},
+                ),
+                "longitude_idx": xr.DataArray(
+                    np.array(self.parser_obj.idx["longitude_idx"]),
+                    dims=("time3"),
+                    coords={"time3": timestamp_array},
+                ),
+            }
+        )
```

Comment:
Sorry I should have noticed this yesterday... I just checked the IDX0 datagram content. I think we should add `ping_number` and `filer_offset` to the stored data as well, since in the manual there are some intricacies in how the pings are associated:
https://www.simrad.online/ek80/interface/ek80_interface_en_a4.pdf#page=162

Also, to make it easy to recognize these, how about prefixing the original data variable names with `idx_`, so there is no confusion of which one is which? i.e. `idx_distance`, `idx_latitude`, etc.